lib/src/realm_object.dart

```diff
@@ -150,28 +150,21 @@ class RealmCoreAccessor implements RealmAccessor {
   }
 
   @override
-  void set(RealmObject object, String name, Object? value, [bool isDefault = false]) {
+  void set(RealmObject object, String name, Object? value, {bool isDefault = false, bool update = false}) {
     final propertyMeta = metadata[name];
     try {
-      if (value is List) {
-        if (value.isEmpty) {
-          return;
-        }
-
-        //This assumes the target list property is empty. `value is List` should happen only when making a RealmObject managed
+      if (value is RealmList<Object>) {
         final handle = realmCore.getListProperty(object, propertyMeta.key);
+        if (update) realmCore.listClear(handle);
```

Comment:
Not sure this is the user intention. We certainly can not make this decision easily. Does the new object with a collection property of one element really mean clear the old list or add this new element to the already existing elements.

Reply:
Yes, the behavior of `addOrUpdate` across strongly-typed SDKs is to give you an object that matches precisely the object you passed. With JS you can achieve some level of merging because it has the notion of undefined, so if you try and update an object `{ a: 1, b: 2 }` with `{ a: 3 }`, you'll end up with `{ a: 3, b: 2}`. In strongly typed languages though, that's not possible using model classes. In the future, we can expose a dictionary-based API for `addOrUpdate` which allows you to only update individual properties.

Regarding merging of collections, I don't think any SDK is doing it and I don't believe there's a predictable way of doing it other than clearing out the existing collection.